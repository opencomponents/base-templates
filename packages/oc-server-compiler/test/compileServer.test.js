jasmine.DEFAULT_TIMEOUT_INTERVAL = 100000;

const _ = require('lodash');
const compileServer = require('../');
const fs = require('fs-extra');
const path = require('path');

const componentPath = path.join(
  __dirname,
  '../../../mocks/handlebars-component'
);
const publishPath = path.join(componentPath, '_compile-server-package');
const publishFileName = 'server.js';

const options = {
  componentPackage: fs.readJSONSync(componentPath + '/package.json'),
  componentPath,
  production: true,
  publishPath,
  publishFileName
};

test('Should correctly compile the server', done => {
  compileServer(options, (err, compiledServerInfo) => {
    expect(err).toBeNull();
    compiledServerInfo.hashKey = 'dummyHashData';
    expect(compiledServerInfo).toMatchSnapshot();
    expect(
      fs.readFileSync(path.join(publishPath, publishFileName), 'UTF8')
    ).toMatchSnapshot();
    fs.removeSync(publishPath);
    done();
  });
});

test('When the server.js compilation fails should return error', done => {
  const options2 = _.cloneDeep(options);
  options2.componentPackage.oc.files.data = 'not-found.js';

  compileServer(options2, err => {
    expect(err).toContain('Entry module not found');
    done();
  });
});

test('When server writing fails should return error', done => {
  const spy = jest
    .spyOn(fs, 'ensureDir')
    .mockImplementation(jest.fn((a, cb) => cb('sorry I failed')));

  compileServer(options, err => {
    expect(err).toMatchSnapshot();
    spy.mockRestore();
    done();
  });
});
