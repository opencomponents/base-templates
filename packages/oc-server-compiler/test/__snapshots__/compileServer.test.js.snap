// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly compile the server 1`] = `
Object {
  "hashKey": "71a1269026ed6dcc29299ed85a8a8bd5660723c3",
  "src": "server.js",
  "type": "node.js",
}
`;

exports[`Should correctly compile the server 2`] = `
"module.exports=function(e){function r(n){if(t[n])return t[n].exports;var u=t[n]={i:n,l:!1,exports:{}};return e[n].call(u.exports,u,u.exports,r),u.l=!0,u.exports}var t={};return r.m=e,r.c=t,r.i=function(e){return e},r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{configurable:!1,enumerable:!0,get:n})},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,\\"a\\",t),t},r.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},r.p=\\"\\",r(r.s=1)}([function(e,r){e.exports=require(\\"lodash\\")},function(e,r,t){\\"use strict\\";Object.defineProperty(r,\\"__esModule\\",{value:!0}),t.d(r,\\"data\\",function(){return n});var n=(t(0),function(e,r){return r(null,{})})}]);
//# sourceMappingURL=server.js.map"
`;

exports[`When server writing fails should return error 1`] = `"sorry I failed"`;
